src/ng/directive/ngModel.js

```diff
@@ -316,6 +318,17 @@ var NgModelController = ['$scope', '$exceptionHandler', '$attrs', '$element', '$
     return isUndefined(value) || value === '' || value === null || value !== value;
   };
 
+  this.$$updateEmptyClasses = function(value) {
+    if (this.$isEmpty(value)) {
```

Comment:
It doesn't really matter because we have control over when it is called but I agree it is probably better for consistency.